cdc/log.cc

```diff
@@ -591,15 +688,21 @@ class transformer final {
             }
         } else {
             // should be insert, update or deletion
+            std::optional<clustering_key> poikey;
             auto process_cells = [&](const row& r, column_kind ckind, const clustering_key& log_ck, std::optional<clustering_key> pikey, const cql3::untyped_result_set_row* pirow) -> std::optional<gc_clock::duration> {
+                if (postimage && !poikey) {
+                    poikey = set_pk_columns(m.key(), ts, tuuid, ++batch_no, res);
+                    set_operation(*poikey, ts, operation::post_image, res);
+                }
                 bytes_opt value;
+                bytes_opt deleted_elements = std::nullopt;
```

Comment:
If I'm not mistaken, this variable can be in smaller scope of the lambda passed to r.for_each_cell in line 597/701

Reply:
Right.